api/go-datakit/transaction.go

```diff
@@ -96,12 +96,15 @@ func (t *transaction) Write(ctx context.Context, path []string, value string) er
 	return nil
 }
 
-// Remove a path (which can either be a key or a directory).
+// Remove deletes a key within the transaction.
 func (t *transaction) Remove(ctx context.Context, path []string) error {
 	p := []string{"branch", t.fromBranch, "transactions", t.newBranch, "rw"}
-	err := t.client.Remove(ctx, path...)
+	for _, dir := range path[0 : len(path)-1] {
```

Comment:
this is indeed probably wrong, we want `len(path)` here. Or just append `path` to `p`.